frontend-web/webclient/app/Applications/Jobs/View.tsx

```diff
@@ -1324,112 +1444,177 @@ function getAppType(job: Job): string {
-                </Button>
-            </Link>
-        )}
-        {appType !== "WEB" || !supportsInterface ? null : (
-            <Link to={redirectToWeb ?? ""} aria-disabled={!redirectToWeb} target={"_blank"}>
-                <Button width="220px" disabled={!redirectToWeb}>
-                    <Icon name={"heroArrowTopRightOnSquare"} />
-                    <div>Open interface {hasMultipleNodes ? `(node ${rank + 1})` : null}</div>
-                </Button>
-            </Link>
+            <Flex>
+                <Link to={`/applications/shell/${job.id}/0?hide-frame`} onClick={e => {
+                    e.preventDefault();
+
+                    const link = findDomAttributeFromAncestors(e.target, "href");
+                    if (!link) return;
+
+                    window.open(
+                        link,
+                        undefined,
+                        "width=800,height=600,status=no"
+                    );
+                }}>
+                    <Button attachedLeft={hasMultipleNodes}>
+                        <Icon name="heroCommandLine" />
+                        <div style={{minWidth: hasMultipleNodes ? "130px" : "164px", maxWidth: "164px"}}>
+                            <Truncate>
+                                Open terminal{hasMultipleNodes ? ` (Node 1)` : null}
+                            </Truncate>
+                        </div>
+                    </Button>
+                </Link>
+                {!hasMultipleNodes ? null :
+                    <RichSelect
+                        items={searchableTerminalLinks}
+                        keys={["searchString"]}
+                        RenderRow={TerminalLinkRow}
+                        FullRenderSelected={TerminalLinkSelectedRow}
+                        onSelect={doNothing}
+                    />
+                }
+            </Flex>
         )}
-        {appType !== "VNC" || !supportsInterface ? null : (
-            <Link to={`/applications/vnc/${job.id}/${rank}?hide-frame`} target={"_blank"} onClick={e => {
-                e.preventDefault();
-
-                const link = findDomAttributeFromAncestors(e.target, "href");
-                if (!link) return;
-
-                window.open(
-                    link,
-                    `vnc-${job.id}-${rank}`,
-                    "width=800,height=450,status=no"
-                );
-            }}>
-                <Button width="220px">
-                    <Icon name={"heroArrowTopRightOnSquare"} />
-                    <div>Open interface {hasMultipleNodes ? `(node ${rank + 1})` : null}</div>
-                </Button>
-            </Link>
+
+        {interfaceLinks.length < 1 ? <></> : (
+            <Flex>
+                <Link to={interfaceLinks[defaultInterfaceId]?.link ?? ""} aria-disabled={!interfaceLinks[defaultInterfaceId]} target={"_blank"}>
+                    <Button attachedLeft={interfaceLinks.length > 1} disabled={!interfaceLinks[0]}>
```

Comment:
Is index 0 the correct index to access here? `defaultInterfaceId` is used around it.